app/models/session.rb

```diff
@@ -12,6 +12,7 @@ class Session < Common::RedisStore
   attribute :token
   attribute :uuid
   attribute :created_at
+  attribute :authn_context
```

Comment:
This addition to the model looks safe...but in the case of rollback I'd like to confirm that the current production code can deserialize a session out of redis that contains this extra attribute. Can probably just test by permuting a serialized redis entry in place with current master code. I'll try this myself and report back.

Reply:
Sorry if i'm stating the obvious, but both `authn_context` and `multifactor` are new, I noticed you're only tagging authn_context. Multifactor I know can't cause issues since it's in no way used in current production.

The backend only uses the current_user.authn_context for serializing to F/E. It uses heuristics and other methods to determine the authn_context (and in the case of rollback those should always be nil). This was a deliberate and mindful decision (to not use the value in current_user on the backend even if it is available.

And I think you should be able to deserialize. Feel free to test. Any arbitrary attributes provided that have no corresponding setter values will be ignored at initialization of the object.

Reply:
Confirmed. 👍